packages/compiler/src/types.ts

```diff
@@ -13,6 +14,13 @@ export interface CompilerInitAction {
   preactVersion: string;
 }
 
+export interface CompilerSetLocalComponentAction {
+  action: 'set-local-components';
+  components: { componentPath: string; componentSource: string }[];
```

Comment:
Can we make this a mapping of path -> source to make its purpose more clear? As an array there's nothing preventing duplicates and setting values is more involved than upserting an entry.

Reply:
For sure! That's a great point. Will refactor that now.